src/workflow/StreamlitUI.py

```diff
@@ -100,14 +126,23 @@ def upload_widget(
 
         # Local file upload option: via directory path
         if st.session_state.location == "local":
-            c2_text, c2_checkbox = c2.columns([2, 1])
+            c2_text, c2_checkbox = c2.columns([1.5, 1], gap="large")
             c2_text.markdown("**OR copy files from local folder**")
-            use_symlink = c2_checkbox.checkbox("Use symlinks", key=f"{key}-symlink", help="Use symbolic links instead of copying files.")
-            with c2.form(f"{key}-local-file-upload"):
-                local_dir = st.text_input(f"path to folder with **{name}** files")
-                if st.form_submit_button(
-                    f"Copy **{name}** files from local folder", use_container_width=True
-                ):
+            use_copy = c2_checkbox.checkbox("Make a copy of files", key=f"{key}-copy_files", value=True, help="Create a copy of files in workspace.")
+            with c2.container(border=True):
+                st_cols = st.columns([0.05, 0.55], gap="small")
+                with st_cols[0]:
+                    st.write("\n")
+                    st.write("\n")
+                    dialog_button = st.button("📁", key='local_browse', help="Browse for your local directory with MS data.", disabled=not TK_AVAILABLE)
+                    if dialog_button:
+                        st.session_state["local_dir"] = self.tk_directory_dialog("Select directory with your MS data", st.session_state["previous_dir"])
+                        st.session_state["previous_dir"] = st.session_state["local_dir"]
+
+                with st_cols[1]:
+                    local_dir = st.text_input(f"path to folder with **{name}** files", value=st.session_state["local_dir"])
+                    
+                if c2.button(f"Copy **{name}** files from local folder", use_container_width=True):
```

Comment:
I added a tkinter directory dialog next to the text input. If the user selects the folder button then it will autofill the text input widget, otherwise the option to manually fill in the text input widget is still available as well.

![image](https://github.com/user-attachments/assets/8ab88ad4-b79b-48da-a699-b96c824add68)

Reply:
Unfortunately this does not work from me (neither on MacOS nor Windows). I get the following error when I try to trigger the file dialog:

`RuntimeError: main thread is not in main loop`

Have you seen this error before?

Reply:
> Unfortunately this does not work from me (neither on MacOS nor Windows). I get the following error when I try to trigger the file dialog:
> 
> `RuntimeError: main thread is not in main loop`
> 
> Have you seen this error before?

Ahh I see, this may occur if there is a delay in the button and it gets pressed multiple times before the Tk dialog is closed. Because it tries to then create another Tk dialog, ontop of the one that already exists which causes the error. I think I will have to add a temporary pause/disable the button until the Tk dialog is closed.

Reply:
Yes thats it! For some reason the file dialog is shown behind my browser window. Hence, I did not see it and pressed the button again.